### ISSUE TITLE
Upgrade to Tesla Version 2023.38.1 - Teslamate shows Model 3 no longer available

### ISSUE THREAD

**Issue by jsight** [teslamate-org/teslamate#3371]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

The vehicle is reported offline. It appears that the drive_state does not have longitude and latitude information.

Example:

```
    "drive_state": {
      "active_route_traffic_minutes_delay": 0,
      "power": 0,
      "shift_state": null,
      "speed": null,
      "timestamp": 1697687194362
    },
```

Seems like it'd be possible to keep tracking charge data at least?

### Relevant log output

```shell
errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
```

### Type of installation

Docker

### Version

1.27.3

**Comment by jsight** [comment]
This works to get the location data: https://owner-api.teslamotors.com/api/1/vehicles/{vehicle_id}/vehicle_data?endpoints=location_data

**Comment by vexingly** [comment]
I can confirm that I'm seeing the same error after the update. The stack trace shows that it is coming from the streaming api though, so I'm not sure if that api is the relevant one?

`  (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4`

**Comment by jsight** [comment]
I'm not as familiar with the streaming api, but it wouldn't be shocking if there are new restrictions there as well.

**Comment by rpearson-catdevnull** [comment]
This is possibly related to the Location Sharing Privacy feature

https://www.notateslaapp.com/software-updates/version/2023.38.1/release-notes

Might be worth double checking Allow Mobile Access is enabled on your vehicle?

**Comment by vexingly** [comment]
Allow Mobile Access is enabled, and the Tesla App has no problem displaying the location.

**Comment by jsight** [comment]
Workaround PR: https://github.com/adriankumpf/teslamate/pull/3373

**Comment by cwanja** [comment]
> Workaround PR: https://github.com/adriankumpf/teslamate/pull/3373
> 
> 

Wonder if that's backwards compatible to other versions, or if it would break their API locations.

**Comment by jsight** [comment]
I suspect not, it should probably choose between them and maybe not make the call at all depending on the API version reported. I don't have a good way to test that and don't know elixir well at all though.

**Comment by jaggerc** [comment]
> > Workaround PR: https://github.com/adriankumpf/teslamate/pull/3373
> > 
> > 
> 
> Wonder if that's backwards compatible to other versions, or if it would break their API locations.  

If you call the endpoint with a vehicle before 2023.38.0 it does not return lat/lon/heading.  It's a complete change with that version and the privacy feature.

**Comment by jsight** [comment]
@seanlw Thanks, I think that is exactly right. I've copied that code more or less verbatim. The only change that I made was to only conditionally make the second request if gps_as_of is nil, as it isn't needed for the other case. This seems to work well.

**Comment by kinteseay** [comment]
I've been using docker to update my teslamate.  Is the only way to apply this patch to use GitHub until a new version is available?  I'm not familiar with the manual process.  Or is there something I can do in docker to get that new file so I can get my model 3 data again.  My model S is still getting data fine.  It is running 2023.32.7 firmware.  So if it asks me to upgrade, should I wait until Teslamate is updated?

**Comment by jsight** [comment]
@kinteseay At the moment, you'd have to build the docker image yourself and point docker-compose to it. If we don't get an official fix, I might try to push this patch to a docker registry to make it easier.

**Comment by jsight** [comment]
It seems like this patch isn't totally working. I'm noticing some missing drives and weird anomalies in the way it is reporting charge sessions. I'm suspicious that it is a side effect of getting 429 errors, but I haven't had time to look into it.

**Comment by PatDal81** [comment]
+1 on that issue with the latest Model 3 update (2023.38.1)

Is this getting traction to get an official fix?  Can't imagine we're the only 2 people with this version on their cars ;)

**Comment by jsight** [comment]
@PatDal81 We aren't the only ones, but it isn't a common version right now. Teslafi stats are showing <1% of cars on this version and the rollout looks like it might have been paused. There's a 2023.38.4 coming out as well, but I haven't heard about any updated release notes.

**Comment by ampersandru** [comment]
surprised Im on 2023.38.1 already when Im usually the last to get updates on my 2018 HW2.5 car. Just mimicking that I also have issues with teslamate communicating with my car. No issues with Teslafi

**Comment by jsight** [comment]
Same, I was really surprised to be in the first wave. I do wonder if this is really just a bug in the vehicle code. I guess we will see when they start pushing the next wave of patches.

**Comment by finnysamuel** [comment]
MYLR - Just upgraded to 2023.38.4. Same error: 
`errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}`
Using the docker image installation. Any help for implementing the workaround is appreciated.

**Comment by thaidavid** [comment]
having the same issue :/

**Comment by maracucho1997** [comment]
Same issue. No drives or charges logged since 2023.38.1.

**Comment by brontide** [comment]
I tried patch #3373 and it appears to resolve this issue, I have closed my duplicate issue #3381 with instructions for how to checkout and use the patch if you have a docker host and docker-compose.  These instructions are likely to be obsolete quickly but are a workaround for now.

**Comment by hollandcloud** [comment]
Same issue here with 2023.38.1 on both vehicles I have (MY with HW3 and MY with HW4). Happy to help test if need be.

**Comment by finnysamuel** [comment]
> @kinteseay At the moment, you'd have to build the docker image yourself and point docker-compose to it. If we don't get an official fix, I might try to push this patch to a docker registry to make it easier.

@jsight Could you push this patch to a registry? Getting errors in my portainer when trying to build the image.

**Comment by seanlw** [comment]
@finnysamuel you can try using my image build `seanlw/teslamate:1.27.3-a` . I've been running this for a few hours and works for both 2023.38.1 and older versions. Not perfect, but works for my needs.

Replace `teslamate/teslamate:latest` with `seanlw/teslamate:1.27.3-a`

**Comment by finnysamuel** [comment]
> > @kinteseay At the moment, you'd have to build the docker image yourself and point docker-compose to it. If we don't get an official fix, I might try to push this patch to a docker registry to make it easier.
> 
> @jsight Could you push this patch to a registry? Getting errors in my portainer when trying to build the image.

Nevermind.. I was able to deploy it in my portainer in Synology NAS.. Thanks a ton for the patch..

**Comment by finnysamuel** [comment]
> @finnysamuel you can try using my image build `seanlw/teslamate:1.27.3-a` . I've been running this for a few hours and works for both 2023.38.1 and older versions. Not perfect, but works for my needs.
> 
> Replace `teslamate/teslamate:latest` with `seanlw/teslamate:1.27.3-a`

Thank you @seanlw - I was able to fix the errors a minute before I saw your comment. I'm sure it will help others.. So big thanks

**Comment by imthefrizzlefry** [comment]
I upgraded MY last night to 2023.38.4, and no response.  unfortunately, I'm running Teslamate as a Home Assistant addon, so it looks like I'm stuck until that gets updated.

**Comment by Macrisu** [comment]
> @finnysamuel you can try using my image build `seanlw/teslamate:1.27.3-a` . I've been running this for a few hours and works for both 2023.38.1 and older versions. Not perfect, but works for my needs.
> 
> Replace `teslamate/teslamate:latest` with `seanlw/teslamate:1.27.3-a`

after that, run:

```
docker-compose pull
docker-compose up -d
```

Thank you for the image

**Comment by jsight** [comment]
I have pushed a build as well: https://hub.docker.com/repository/docker/jsightler/teslamate/general

This is based on this PR: https://github.com/adriankumpf/teslamate/pull/3373

My previous version broke some things as I misunderstood elixir pattern matching. I think this one might be a little better.

**Comment by brontide** [comment]
Thanks @jsight, I ran the updated code and it seems cleaner from the logged output.

**Comment by Macrisu** [comment]
docker pull jsightler/teslamate:20231029

> I have pushed a build as well: https://hub.docker.com/repository/docker/jsightler/teslamate/general
> 
> This is based on this PR: #3373
> 
> My previous version broke some things as I misunderstood elixir pattern matching. I think this one might be a little better.

docker pull jsightler/teslamate:20231029

**Comment by seanlw** [comment]
@jsight, nice work. I can confirm it's also backwards compatible.

**Comment by boons8895** [comment]
> @jsight would you mind creating an image for other architectures as well? I cant run the amd64 image on the armv7 architecture of my pi2.

Yeah I ran into this issue as well with my Pi 4

**Comment by PatDal81** [comment]
Instead of creating forks (nothing against you jsight), wouldn't it be simpler to push this into the official build and have an official release?  I get the feeling more and more people are updating to this version.

**Comment by imthefrizzlefry** [comment]
I can agree, I would love to see this update merged to a new release.  In the meantime, at least I'm still collecting data with the fork.

**Comment by brontide** [comment]
> Instead of creating forks (nothing against you jsight), wouldn't it be simpler to push this into the official build and have an official release? I get the feeling more and more people are updating to this version.

@jsight has created the patch and offered it to Teslamate via a pull request, it's up to the project owner/maintainer to accept the patch and then create a new release based on it. 

For those with alternate arches, you can follow the instructions in #3381 to build directly from the alternate code, this is what I am doing for the time being.

**Comment by lcorreia007** [comment]
I have a model s and I notice that after the update to 2023.38.4 not just teslamate stop updating data but it start draining my battery...had to stop teslamate completely to fix the vampire drain.
Is anyone experience the same behavior?

**Comment by marcbaier** [comment]
Any news on when this will be integrated into a new release of Teslamate ?

**Comment by tesla2021a** [comment]
Hi, I unfortunately upgraded 2023.38.6 and now plagued by this bug :-(

I run Teslamate on a RaspberryPie and the docker image seanlw/teslamate:1.27.3-a and jsightler/teslamate:20231029 do not work for me.

seanlw/teslamate:1.27.3-a won't even download, giving error: 
```ERROR: for teslamate  no matching manifest for linux/arm/v7 in the manifest list entries```

jsightler/teslamate:20231029 do download but gives the following error when starting
```teslamate_teslamate_1 exited with code 1```
```teslamate_1  | standard_init_linux.go:219: exec user process caused: exec format error```

Somebody has a solution for someone running docker on a RaspberryPie running Raspbian GNU/Linux 11 (bullseye)?

Thanks.

**Comment by tesla2021a** [comment]
> @finnysamuel you can try using my image build `seanlw/teslamate:1.27.3-a` . I've been running this for a few hours and works for both 2023.38.1 and older versions. Not perfect, but works for my needs.
> 
> Replace `teslamate/teslamate:latest` with `seanlw/teslamate:1.27.3-a`

For what distribution did you release it? I can't get it to work on my RaspberryPie running Raspbian GNU/Linux 11 (bullseye), I get the error `ERROR: for teslamate  no matching manifest for linux/arm/v7 in the manifest list entries`

**Comment by jolcese** [comment]
`amd64` and `arm64`. No arm v7
<img width="1210" alt="image" src="https://github.com/adriankumpf/teslamate/assets/3219544/cb50d796-6f6f-442d-8bc4-bf997ebb7ae0">

**Comment by imthefrizzlefry** [comment]
In case anyone cares, I already had the `jsightler/teslamate:20231029` image running after updating my car to 2023.38.4, and updating to 2023.38.6 just now did not appear to break anything for me.  

However, I did see a red exclamation indicating "Health Check Failed" in the corner, but I don't really know if that was appearing before.  I mostly use Grafana for everything.

**Comment by tesla2021a** [comment]
> `amd64` and `arm64`. No arm v7. <img alt="image" width="1210" src="https://user-images.githubusercontent.com/3219544/280933413-cb50d796-6f6f-442d-8bc4-bf997ebb7ae0.png">
> 
> You need to use the 64bits version of the OS

Shit, because of other stuff running on that little guy, I'm stuck at 32bits. I guess I'll see if I can shuffle some stuff around between my different pie...

**Comment by chr1s-code** [comment]
> > `amd64` and `arm64`. No arm v7. <img alt="image" width="1210" src="https://user-images.githubusercontent.com/3219544/280933413-cb50d796-6f6f-442d-8bc4-bf997ebb7ae0.png">
> 
> > 
> 
> > You need to use the 64bits version of the OS
> 
> 
> 
> Shit, because of other stuff running on that little guy, I'm stuck at 32bits. I guess I'll see if I can shuffle some stuff around between my different pie...

Just build the image yourself on the pi. After that select the tag of your image and it should work fine. 
I'm currently running @jsight's patch on my Pi 2B which has an armv7 CPU as well.

**Comment by tesla2021a** [comment]
Tried to build an image but the Teslamate doc for Debian is outdated and non functional on RaspberryPie running bookworm. grafana failed to install and had to follow the instruction on the grafana website to make it work. node.js 14.x is deprecated and fails to install on bookworm :-(

I guess until Adrian decides to fix the problem in his installation or someone publishes a fixed teslamate docker file compatible with a RaspberryPie, I'll install teslamate docker on my PC and point my reverse proxy to it instead of my Raspberry Pie. Wasted my morning on this :-( Such is life when using Linux...

**Comment by JakobLichterfeld** [comment]
At the moment it looks like we have to fork the project and run a GitHub Cli worker that builds the new Docker image for all platforms with buildx, maybe I'll have time to do that in the next few days,

**Comment by tesla2021a** [comment]
> At the moment it looks like we have to fork the project and run a GitHub Cli worker that builds the new Docker image for all platforms with buildx, maybe I'll have time to do that in the next few days

Thanks. I'm up an running again but on my Windows laptop instead of my RaspberryPie. Not the most practical solution but at least, it works.

I saw through Teslafi that 2023.38.6 is being pushed to more people today. I guess an influx of users complaining it ain't working anymore will start to show up soon. Having a solution that doesn't require a rebuild on something else will be welcomed. Thanks again. I whish Adrian would come back soon but we've seen nothing from him for a few weeks regarding this problem :-(

**Comment by jsight** [comment]
If someone wants to setup a fork using an organization, I'd be happy to help when I have time. I do wonder if it'd be a bad idea to reuse the same name, though.

**Comment by jsight** [comment]
Yeah, that's possible, but we'd want @adriankumpf's permission to have the same name. Also, how do you setup a free org  for FOSS in docker hub now? I'm not seeing the option at the moment.

**Comment by WarpSpeedo** [comment]
neither of the 2 fixes i've seen on this thread seem to work on my pi.  I didn't realize that they were not written for the Pi... do we know if @adriankumpf abandoned the project?

**Comment by samyun** [comment]
![image](https://github.com/adriankumpf/teslamate/assets/4411707/ccf8763a-2cf5-493a-ab07-12e44a977a4e)
Adrian when he sees this thread

My guess is he's on vacation or something, since his profile doesn't show any activity for November.

**Comment by WiesiDeluxe** [comment]
same issue on my MYP after update to 2023.38.6 - with jsightler/teslamate:20231029 it seems to work again, thanks for that, hopefully official fix is coming fast :)

**Comment by JakobLichterfeld** [comment]
> At the moment it looks like we have to fork the project and run a GitHub Cli worker that builds the new Docker image for all platforms with buildx, maybe I'll have time to do that in the next few days

Invested 30 Minutes, first build seems fine, waiting for the runners to finish: https://github.com/teslamate-fork/teslamate-fork/actions/runs/6816208439 --> https://hub.docker.com/r/teslamatefork/teslamatefork

**Comment by kkthxbye-code** [comment]
@JakobLichterfeld I think you need to follow step one here for buildjet runners to work:

https://buildjet.com/for-github-actions/docs/getting-started/run-your-first-workflow#1-give-buildjet-access-to-your-git-hub-repository

Both the arm flows use buildjet:

https://github.com/teslamate-fork/teslamate-fork/blob/192c7c3938c0e0fde09517c1a2cedb6e653c212c/.github/workflows/buildx.yml#L24-L29

**Comment by JakobLichterfeld** [comment]
Absolutely right, missed that and fixed

**Comment by JakobLichterfeld** [comment]
And working for all platforms, will test tomorrow, only on mobile atm

**Comment by tesla2021a** [comment]
teslamatefork works for me. I only replaced
`image: teslamate/teslamate:latest`
with
`image: teslamatefork/teslamatefork:edge`
in my docker_compose.yml file and once `sudo docker compose up` was ran, system came back to life on my RasberryPie 4 running bookworm 64bits.

**Comment by JakobLichterfeld** [comment]
Yeah, so it is working
In docker-compose change to `teslamatefork/teslamatefork:latest`, then `docker compose up -d`and enjoy for the time being.

**Comment by rutger1140** [comment]
Can confirm this teslamatefork works (amd64). 

Was about to switch to Tessie to safe myself the hassle of maintaining the install, but the extensible feature set of TeslaMate is unparalleled. 
(Lack of dynamic tariffs on Tessie is holding me back.)

Hopefully no more drastic Tesla API changes in the near future 🙏

**Comment by golfleep** [comment]
> Yeah, so it is working In docker-compose change to `teslamatefork/teslamatefork:edge`, then `docker compose up -d` and enjoy for the time being.

Thanks for this! Back up and running after an unintentional upgrade to 2023.38.6

**Comment by Techtobi83** [comment]
Good evening, is anybody able to help a n00b like me?
Long story short, I was already having trouble understanding how to install Teslamate on my Synology in 2022. I found the following helpful instructions, which I followed 1:1 and Teslamate has been running unchanged and reliably for me since 2022:

Link to "the idiots guide":

**Comment by jolcese** [comment]
> Good evening, is anybody able to help a n00b like me? Long story short, I was already having trouble understanding how to install Teslamate on my Synology in 2022. I found the following helpful instructions, which I followed 1:1 and Teslamate has been running unchanged and reliably for me since 2022:
> 
> Link to "the idiots guide": #113
> 
> My docker-compose.yml from 2022 is still unchanged in my backups. But unfortunately I don't understand at all what I have to do to get your workaround to work. I've been learning how to use Docker for the last 2 years, but everything via GUI within the Synology operating system.
> 
> It would be really cool if someone could help me :-)

Replace on your docker `docker-compose.yml` 
`image: teslamate/teslamate:latest`
by 
`image: teslamate-fork/teslamate-fork:edge`

**Comment by Techtobi83** [comment]
> Replace on your docker `docker-compose.yml`
> 
> `image: teslamate/teslamate:latest` by `image: teslamatefork/teslamatefork:edge`

Okay, then upload and move the file to same location like 2022 as explained in "idiots guide" and then? 

simply "docker-compose up -d" and thats it? 

(another problem i may have: i do not have any known option (for me) to make an isolated backup from existing installation of teslamate. i only have full backups from the complete synology. all my other docker-container were managed during GUI and they have own folders i can copy & paste)

**Comment by djreedoo** [comment]
Hi,Yes indeed if anybody could write a short tutorial explaining how to switch from an installed TeslaMate to this ex fork.Would be greatly appreciated 👍🏻regards Le 10 nov. 2023 18:54, Techtobi83 ***@***.***> a écrit :
Good evening, is anybody able to help a n00b like me?
Long story short, I was already having trouble understanding how to install Teslamate on my Synology in 2022. I found the following helpful instructions, which I followed 1:1 and Teslamate has been running unchanged and reliably for me since 2022:
Link to "the idiots guide":

**Comment by jolcese** [comment]
> > Replace on your docker `docker-compose.yml`
> > `image: teslamate/teslamate:latest` by `image: teslamatefork/teslamatefork:edge`
> 
> Okay, then upload and move the file to same location like 2022 as explained in "idiots guide" and then?
> 
> simply "docker-compose up -d" and thats it?
> 
> (another problem i may have: i do not have any known option (for me) to make an isolated backup from existing installation of teslamate. i only have full backups from the complete synology. all my other docker-container were managed during GUI and they have own folders i can copy & paste)

Yes, upload the file and move it. If you feel comfortable, you can directly modify the file that exists on `/volume1/@docker/docker-compose.yml` 
And finally `docker-compose up -d` to restart the containers

**Comment by jolcese** [comment]
> Hi,Yes indeed if anybody could write a short tutorial explaining how to switch from an installed TeslaMate to this ex fork.Would be greatly appreciated 👍🏻regards Le 10 nov. 2023 18:54, Techtobi83 ***@***.***> a écrit : Good evening, is anybody able to help a n00b like me? Long story short, I was already having trouble understanding how to install Teslamate on my Synology in 2022. I found the following helpful instructions, which I followed 1:1 and Teslamate has been running unchanged and reliably for me since 2022: Link to "the idiots guide": —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

It will depend if you're running as docker (just requires to change the image) or if you're manually installing it (will require either manually patching the source code like this: https://github.com/adriankumpf/teslamate/pull/3373/files or using a fork like the one here: https://github.com/teslamate-fork/teslamate-fork

**Comment by JakobLichterfeld** [comment]
> Hi,Yes indeed if anybody could write a short tutorial explaining how to switch from an installed TeslaMate to this ex fork.

Already done, see:
https://github.com/teslamate-fork/teslamate-fork#installation-of-the-fork-with-the-fix-for-tesla-fw-2023381

**Comment by alexeiw123** [comment]
> > Hi,Yes indeed if anybody could write a short tutorial explaining how to switch from an installed TeslaMate to this ex fork.
> 
> Already done, see: https://github.com/teslamate-fork/teslamate-fork#installation-of-the-fork-with-the-fix-for-tesla-fw-2023381

Some simple questions from a power user but developer noob...

- If I switch to this fork, is this a risky security decision given my vehicle credentials are involved in this software?
- Does this work on rasp pi 3b+ (I think I used a 64 bit image)?

I'd like to thank my wife for pressing the update button for the first time _ever_. I was planning to leave this one for a while!

**Comment by jsight** [comment]
I don't think it is any more or less risky than it was before, tbh. Then again, I was involved in tweaking the code, so I would say that! :) But in general, I don't see it as more risky than any other average github project.

**Comment by cwanja** [comment]
> If I switch to this fork, is this a risky security decision given my vehicle credentials are involved in this software?

The base is still Adrian's repo. It's slightly modified to pull from the updated Tesla API endpoint.

**Comment by alexeiw123** [comment]
Thankyou both.
I guess my risk is mitigated too as I've shared my vehicle to a secondary account for teslamate so I can easily pull access from my main count at anytime.

Will give it a look a little later.

**Comment by alexeiw123** [comment]
> * Does this work on rasp pi 3b+ (I think I used a 64 bit image)?

for others, the answer is yes.

**Comment by tesla2021a** [comment]
And activate MFA for added security. Your token is useless until acknowledged by an MFA session.

**Comment by alexeiw123** [comment]
Absolutely, that goes without saying! MFA all the things!

**Comment by Techtobi83** [comment]
Moin aus dem Sauerland.... oh sorry, to stay international: good morning everyone

i followed @jolcese instructions above and now everything looks like it is working again
![Screenshot 2023-11-11 065904](https://github.com/adriankumpf/teslamate/assets/86073272/c39e1eee-d09c-4072-b6e4-678a5ea16280)

At this moment, my Tesla is updating to x.38.6. Will do a short trip and see if the tour will be logged by Teslamate as usual

**Comment by JakobLichterfeld** [comment]
> > * Does this work on rasp pi 3b+ (I think I used a 64 bit image)?
> 
> for others, the answer is yes.

And on 32bit build (armv7) as well, so all platforms original teslamate is build for.

**Comment by JakobLichterfeld** [comment]
> If I switch to this fork, is this a risky security decision given my vehicle credentials are involved in this software?

That's why I build the fork with automatic docker image generation to have a transparent process and you can see what is build

**Comment by Goger-TKL** [comment]
Thanks for the fork, solved the problem for me. I sthere any way to get informed when the fix is integrated in the main fork to switch back then?

**Comment by on1krr** [comment]
Hi.  first, thanks to those that are working this issue on the fork.  

My question is that I see from following this thread, jakobLitcherfeld has posted instructions on  how to move to the forked docker container.  My problem is that I am running a native build and am a complete novice with git being a hardware designer by trade.  I have googled on how to switch git to a fork but cannot be sure what I am reading is correct.  Could someone please provide a method on how to repoint a cloned testmate instance (manual install instructions from Adrian)  to the testmate-fork repository?

Thanks in advance.

**Comment by JakobLichterfeld** [comment]
> Thanks for the fork, solved the problem for me. Is there any way to get informed when the fix is integrated in the main repository to switch back then?

Follow #3371 and/or follow the original repo with email for each new release

**Comment by Techtobi83** [comment]
> At this moment, my Tesla is updating to x.38.6. Will do a short trip and see if the tour will be logged by Teslamate as usual

After updating Tesla EV, everything works great. Thanks guys for your help.

**Comment by marcbaier** [comment]
> Moin aus dem Sauerland.... oh sorry, to stay international: good morning everyone
> 
> i followed @jolcese instructions above and now everything looks like it is working again ![Screenshot 2023-11-11 065904](https://user-images.githubusercontent.com/86073272/282222528-c39e1eee-d09c-4072-b6e4-678a5ea16280.png)
> 
> At this moment, my Tesla is updating to x.38.6. Will do a short trip and see if the tour will be logged by Teslamate as usual

Hi I also followed the instructions and it at least looks like I did everything correctly: 
![Screen Shot 2023-11-11 at 14 33 13 PM](https://github.com/adriankumpf/teslamate/assets/97667985/b5b93a05-9dcd-436b-847b-1fb2562ec836)
But when I start Teslamate, it still shows the "old" data with the old firmware. Do I need to "trigger "something in order to get new data ? 
![Screen Shot 2023-11-11 at 14 35 48 PM](https://github.com/adriankumpf/teslamate/assets/97667985/420d255b-cdd3-4b9d-bb1a-36bb082a4b85)
Thanks so much for your help !

**Comment by marcbaier** [comment]
It's working now. Had to wake up the car :) Now the new data is showing just fine. This community is great !

**Comment by cleanev** [comment]
> Yeah, so it is working In docker-compose change to `teslamatefork/teslamatefork:edge`, then `docker compose up -d` and enjoy for the time being.

Running on Pi4b+ on 64bit Raspberry Pi OS (bullseye)
As things are running normally, unless I get logged out Settings page will not be a problem. Looking to see if anyone else has these challenges and how they fixed them, if any
1. Battery Health Dashboard complains ‘Dashboard not found’ 
2. Settings on main Teslamate page lands on ‘Internal Server Error’ with below errors from teslamate docker log

Request: GET /settings
** (exit) an exception was raised:
    ** (KeyError) key :id not found in: nil
If you are using the dot syntax, such as map.field, make sure the left-hand side of the dot is a map
        (teslamate 1.28.0-dev) lib/teslamate_web/live/settings_live/index.ex:183: anonymous fn/2 in TeslaMateWeb.SettingsLive.Index.prepare/1
        (elixir 1.15.7) lib/enum.ex:2510: Enum."-reduce/3-lists^foldl/2-0-"/3
        (teslamate 1.28.0-dev) lib/teslamate_web/live/settings_live/index.ex:15: TeslaMateWeb.SettingsLive.Index.mount/3
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/utils.ex:301: anonymous fn/6 in Phoenix.LiveView.Utils.maybe_call_live_view_mount!/5
        (telemetry 1.2.1) /opt/app/deps/telemetry/src/telemetry.erl:321: :telemetry.span/3
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/static.ex:270: Phoenix.LiveView.Static.call_mount_and_handle_params!/5
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/static.ex:110: Phoenix.LiveView.Static.render/3
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/controller.ex:39: Phoenix.LiveView.Controller.live_render/3
2023-11-11 14:27:40.900 car_id=1 [info] Start / :online
2023-11-11 14:27:40.960 car_id=1 [info] Connecting ...
2023-11-11 14:28:38.127 [info] GET /settings
2023-11-11 14:28:38.154 [info] Sent 500 in 26ms
2023-11-11 14:28:38.156 [error] #PID<0.2738.0> running TeslaMateWeb.Endpoint (connection #PID<0.2737.0>, stream id 1) terminated
Server: 10.2.2.7:4000 (http)

**Comment by vzhivkov** [comment]
I'm trying to update with the given fork docker image. I changed the yml file and ran the command several times but I keep getting errors every time.

![image](https://github.com/adriankumpf/teslamate/assets/7021418/3ab192a4-ac58-4b83-9d9c-3a8d17f7e0c0)

I even undid the changes in the yml file to see if it will give some error but it didn't. The Raspberry Pi has wired internet connection with no issues that I can see. What might cause such errors?

**Comment by jolcese** [comment]
> I'm trying to update with the given fork docker image. I changed the yml file and ran the command several times but I keep getting errors every time.
> 
> ![image](https://user-images.githubusercontent.com/7021418/282254328-3ab192a4-ac58-4b83-9d9c-3a8d17f7e0c0.png)
> 
> I even undid the changes in the yml file to see if it will give some error but it didn't. The Raspberry Pi has wired internet connection with no issues that I can see. What might cause such errors?

The docker registry is currently down: https://www.dockerstatus.com/

**Comment by vzhivkov** [comment]
> > I'm trying to update with the given fork docker image. I changed the yml file and ran the command several times but I keep getting errors every time.
> > ![image](https://user-images.githubusercontent.com/7021418/282254328-3ab192a4-ac58-4b83-9d9c-3a8d17f7e0c0.png)
> > I even undid the changes in the yml file to see if it will give some error but it didn't. The Raspberry Pi has wired internet connection with no issues that I can see. What might cause such errors?
> 
> The docker registry is currently down: https://www.dockerstatus.com/

Didn't even thought it might be something like this. Thanks. I'll check and try when it's back to normal state.

**Comment by GBouerat** [comment]
This worked for me : https://github.com/teslamate-fork/teslamate-fork#installation-of-the-fork-with-the-fix-for-tesla-fw-2023381
Tesla version: 2023.38.6
thanks 👍

**Comment by djreedoo** [comment]
Hi,
I made the changes to my Raspberry PI 2 and Oracle Cloud and they both work despite the update.

Thanks very much!

**Comment by NickWhitNTES** [comment]
I was fighting the docker registry being down!  TIMING!

Anyway, works now, thanks!

**Comment by tomsteenbakkers** [comment]
Hi,

I made the changes (`image:teslamatefork:edge`) on my TransIP server running Teslamate and it’s working again.

Thanks very much!

**Comment by bekovan** [comment]
Working great ! Thanks for investing the time to solve the issue.

**Comment by DumSkidderik** [comment]
> @finnysamuel you can try using my image build `seanlw/teslamate:1.27.3-a` . I've been running this for a few hours and works for both 2023.38.1 and older versions. Not perfect, but works for my needs.
> 
> Replace `teslamate/teslamate:latest` with `seanlw/teslamate:1.27.3-a`

Perfect, thanks Sean!

**Comment by Techtobi83** [comment]
After changing to 

> > At this moment, my Tesla is updating to x.38.6. Will do a short trip and see if the tour will be logged by Teslamate as usual
> 
> After updating Tesla EV, everything works great. Thanks guys for your help.

I noticed another small problem: While Teslamate itssef is working fine, i get no actual values via MQTT from my teslamate instance.  At this moment i do not know the reason for this, because IP:Port (Docker-Container) did not change. 

Is it possible, that teslamatefork/teslamatefork:edge does not include old stuff like mqtt-support?

**Comment by ahave2** [comment]
> I noticed another small problem: While Teslamate itssef is working fine, i get no actual values via MQTT from my teslamate instance. At this moment i do not know the reason for this, because IP:Port (Docker-Container) did not change.
> 
> Is it possible, that teslamatefork/teslamatefork:edge does not include old stuff like mqtt-support?

No, I make use of MQTT and I changed my docker to teslamatefork a few days ago with no problems.

**Comment by Techtobi83** [comment]
Ah, oh my god, found the (my!) issue: 

>   mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    # ports:
    #   - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

I had to delete the "#" out of the yml-file, no idea why they were still in my old configuration file from 2022. (may be i fixed it in the past via Docker-GUI and edited manually the mosquitto-container  by adding the ports)

**Comment by cwanja** [comment]
This is resolved in the latest TeslaMate release 1e7c9a4. Alternatively, could keep your fork and merge future updates to your own fork.

**Comment by Dulanic** [comment]
Closing as resolved /w https://github.com/teslamate-org/teslamate/releases/tag/v1.27.4
